CDEFramework/Libraries/LegacyDictionarySupport/GlossaryAPIClient.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Net.Http.Formatting;
+using Common.Logging;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+using LegacyDictionarySupport.Configuration;
+
+namespace LegacyDictionarySupport
+{
+    public class GlossaryAPIClient
+    {
+        static ILog log = LogManager.GetLogger(typeof(GlossaryAPIClient));
+
+        private HttpClient _client = null;
+
+        /// <summary>
+        /// Creates a new instance of a Glossary API client.
+        /// </summary>
+        /// <param name="client">An HttpClient that has the BaseAddress set to the API address.</param>
+        public GlossaryAPIClient()
+        {
+            string baseApiPath = WebAPISection.GetAPIUrl();
+            string appPath = ConfigurationManager.AppSettings["GlossaryAPIAppPath"];
+            string versionPath = ConfigurationManager.AppSettings["GlossaryAPIVersionPath"];
+
+            if (string.IsNullOrWhiteSpace(appPath))
+                throw new ConfigurationErrorsException("error: GlossaryAPIAppPath cannot be null or empty");
+
+            if (string.IsNullOrWhiteSpace(versionPath))
+                throw new ConfigurationErrorsException("error: GlossaryAPIVersionPath cannot be null or empty");
+
+            HttpClient client = new HttpClient();
+            //NOTE: the base URL MUST have a trailing slash
+            client.BaseAddress = new Uri(String.Format("{0}/{1}/{2}/", baseApiPath, appPath, versionPath));
```

Comment:
Similar question to the one in `GlossaryAPIClient`.  Why not retrieve the full URL instead of building it?  That way you can avoid configuration mistakes due to a value having, or not having a slash character.

Reply:
Again, this is how we set up the API client for CTS back in the day, so I was just going with consistency. It can be changed to retrieving the full URL if that's what you and @bryanpizzillo think is best now!